metrics-exporters/leadtime-exporter/lt-exporter.py

```diff
@@ -0,0 +1,141 @@
+import yaml
+import json
+import time
+import urllib3
+import requests
+import os
+from datetime import datetime, timezone
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+from prometheus_client import start_http_server
+from prometheus_client.core import CounterMetricFamily,InfoMetricFamily,GaugeMetricFamily, REGISTRY
+
+class CommitCollector(object):
+    _prefix = "https://api.github.com/repos/"
+    _suffix = "/commits"
+    def __init__(self, username, token, projects, apps):
+        self._username = username
+        self._token = token
+        self._projects = projects
+        self._apps = apps
+    def collect(self):
+        ld_metric = GaugeMetricFamily('github_commit_timestamp', 'Commit timestamp', labels=['namespace', 'app', 'build', 'commit_hash'])
+        ld_metrics = generate_ld_metrics_list(projects)
+        for my_metric in ld_metrics:
+            print("Namespace: ", my_metric.namespace, ", App: ", my_metric.app_name, ", Build: ", my_metric.build_name)
+            ld_metric.add_metric([my_metric.namespace, my_metric.app_name, my_metric.build_name, my_metric.commit_hash], my_metric.lead_time)
+            yield ld_metric
+class mdt_metric:
+    def __init__(self, app_name):
+        self.app_name = app_name
+        self.labels = None
+        self.repo_url = None
+        self.commiter = None
+        self.commit_hash = None
+        self.commit_timestamp = None
+        self.build_name = None
+        self.build_config_name = None
+        self.image_location = None
+        self.image_name = None
+        self.image_tag = None
+        self.image_hash = None
+        self.deployment_config_name = None
+        self.deployment_timestamp = None
+        self.lead_time = 0
+    def getCommitTime(self):
+        _prefix = "https://api.github.com/repos/"
+        _suffix = "/commits/"
+        myurl = self.repo_url
+        url_tokens = myurl.split("/")
+        url = _prefix + url_tokens[3] + "/" +url_tokens[4].split(".")[0] +_suffix+self.commit_hash
+        response = requests.get(url, auth=(username, token))
```

Comment:
Some logging around requests would be useful for any errors we are getting around this metric.

Reply:
We're not expecting errors in connectivity, and I don't think logging responses will be useful, we may add a try/expect block to log connectivity errors, What do you think?